timesketch/api/v1/resources/archive.py

```diff
@@ -424,96 +442,364 @@ def _export_view(self, view, sketch, zip_file):
         """
-        if sketch.get_status.status == "archived":
+        logger.info("Unarchiving sketch %s ('%s').", sketch.id, sketch.name)
+
+        # 1. Pre-flight checks
+        if sketch.get_status.status != "archived":
             abort(
                 HTTP_STATUS_CODE_BAD_REQUEST,
-                "Unable to archive a sketch that was already archived "
-                "(sketch: {:d})".format(sketch.id),
+                "Unable to unarchive a sketch that wasn't already archived "
+                f"(sketch: {sketch.id})",
             )
 
-        labels_to_prevent_deletion = current_app.config.get(
-            "LABELS_TO_PREVENT_DELETION", []
-        )
+        errors_occurred = False
+        error_details = []
 
-        for label in labels_to_prevent_deletion:
-            if sketch.has_label(label):
-                abort(
-                    HTTP_STATUS_CODE_FORBIDDEN,
-                    "A sketch with the label {:s} cannot be " "archived.".format(label),
+        # Check if any timeline is in a state that would prevents unarchiving.
+        # TODO enforce after 2026-01-01
+        # https://github.com/google/timesketch/issues/3518
+        statuses_preventing_unarchival = ["processing", "fail"]
+        for timeline in sketch.timelines:
+            timeline_status = timeline.get_status.status
+            if timeline_status in statuses_preventing_unarchival:
+                base_warning_msg = (
+                    f"Unarchiving sketch {sketch.id}, but it contains timeline "
+                    f"'{timeline.name}' (ID: {timeline.id}) in a '{timeline_status}' "
+                    "state. "
                 )
+                specific_advice = ""
+                if timeline_status == "fail":
+                    specific_advice = (
+                        "It is recommended to fix this timeline (e.g., by deleting it) "
+                        "because the sketch cannot be archived again in this state."
+                    )
+                elif timeline_status == "processing":
+                    specific_advice = (
+                        "Please wait for it to finish processing. If it seems to be "
+                        "stuck, contact your system administrator to resolve the issue."
+                    )
+                    # only a processing timeline is stopping the flow
+                    errors_occurred = True
+
+                general_advice = " You can use 'tsctl find-inconsistent-archives' to find such sketches."  # pylint: disable=line-too-long
+                warning_msg = f"{base_warning_msg}{specific_advice}{general_advice}"
+                error_details.append(warning_msg)
+                logger.warning(warning_msg)
+
+        if errors_occurred:
+            logger.error(
+                "Unarchiving sketch %s failed because one or more indices could not "
+                "be opened. Errors: %s",
+                sketch.id,
+                "; ".join(error_details),
```

Comment:
This `error_details` contains potential sensitive case details `timeline.name`.
Also: You already send each of the warning to the log above in line 699. So do you really need this again as error?